### PR TITLE
Use modern C++ constructs in EventSetup

### PR DESCRIPTION
**Pull request by Dr15Jones**
This is a refactoring to improve the overall code quality of part of the EventSetup.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
please test

**Comment by Dr15Jones** [comment]
+1

File: FWCore/Framework/interface/Callback.h [hunk 1]
```diff
@@ -47,12 +48,12 @@ namespace edm {
                              =CallbackSimpleDecorator<TRecord> >
       class Callback : public CallbackBase {
        public:
-         typedef TReturn (T ::* method_type)(const TRecord&);
+         using  method_type = TReturn (T ::*)(const TRecord&);
```
Comment (fabiocos) [inline FWCore/Framework/interface/Callback.h]: @Dr15Jones for my own education, is there a functional difference between these two syntaxes, or is it just aesthetics? Browsing around https://en.cppreference.com/w/cpp/language/type_alias I would say the latter, but I would like to know your motivation

Comment (Dr15Jones) [reply FWCore/Framework/interface/Callback.h]: The `using` directive can do everything the `typedef` can do and then even more. The modern C++ community seems to favor deprecating the use of `typedef` in favor of the more flexible `using`. That is why I've been converting code from using one to the other.

**Comment by Dr15Jones** [comment]
@fabiocos asked
> do we expect changes because of this PR in the DQM output?

This should not have changed DQM. However, my change to similar code last week also caused a series of DQM plots to be slightly different. We ultimately decided that was because those plots are not always reproducible. I'm assuming the same thing happened here.

**Comment by Dr15Jones** [comment]
One such difference, is 1 Tau more or less tau:

https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_10_3_X_2018-08-12-1100+24270/27959/10224.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017PU_GenSimFull+DigiFullPU_2017PU+RecoFullPU_2017PU+HARVESTFullPU_2017PU/L1T_L1TStage2CaloLayer2_NonIsolated-Tau.html

If this code change actually was the cause of a problem then I would have expected drastic problems (crashes, completely missing objects, etc) not something this subtle.

**Comment by fabiocos** [comment]
+1

@Dr15Jones thank you for the check

**Pull request merged**